Dynatrace/dynatrace-operator
# Add hostMappings on EdgeConnect provisioning via API

## Description

[K8S-10097](https://dt-rnd.atlassian.net/browse/K8S-10097)

Add `hostMappings` to JSON body when creating EdgeConnect object via DT API.

Only the Kubernetes Automation host pattern must be added to this field, forwarding to the K8s default DNS:
```
...
  “hostMappings”: [ 
    {
      “from”: “my-edgeconnect.dynatrace.a273ec656-603d-46c8-b5f5-5c47a6903dff.kubernetes-automation”, 
      “to”: “kubernetes.default.svc.cluster.local” 
    } 
  ],
...
```

(!) Currently, the API supports posting an object with this new property; [you can see it in the schema](https://dib18963.dev.apps.dynatracelabs.com/platform/swagger-ui/index.html?urls.primaryName=AppEngine%20-%20EdgeConnect#/EdgeConnect/createEdgeConnect). But it doesn't return it in the response, so we're not 100% sure it's getting stored (we trust the API for now)

## How can this be tested?

- Deploy Operator
- Apply EdgeConnect with `spec.KubernetesAutomation.enabled: true` and `provisioner: true`
- No errors are shown in operator logs
- New EdgeConnect is provisioned to the tenant, check it here:
`https://<your-tenant>.dev.apps.dynatracelabs.com/ui/apps/dynatrace.edgeconnect.management/`
- You can also test getting the provisioned EdgeConnect with a GET request to the API
`https://<your-tenant>.dev.apps.dynatracelabs.com/platform/app-engine/edge-connect/v1/edge-connects/<edge-connect-unique-id>`
But as mentioned, the hostMappings property is not added in the response... So we can't really test that it is stored.
